easypermissions/src/main/java/pub/devrel/easypermissions/helper/PermissionHelper.java

```diff
@@ -0,0 +1,126 @@
+package pub.devrel.easypermissions.helper;
+
+import android.app.Activity;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.StringRes;
+import android.support.v4.app.Fragment;
+
+import java.util.List;
+
+import pub.devrel.easypermissions.RationaleDialogFragment;
+
+/**
+ * Delegate class to make permission calls based on the 'host' (Fragment, Activity, etc).
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+public abstract class PermissionHelper<T> {
+
+    private static final String TAG = "PermissionHelper";
+
+    private T mHost;
+
+    @NonNull
+    public static PermissionHelper getInstance(Object host) {
```

Comment:
This method name confused before I looked at its implementation: I think it should be called `newInstance`. Otherwise, it looks like you're getting a single `PermissionHelper` instance per `Activity`/`Fragment` (which set off alarm bells in my mind about memory leaks 🙀)